### DISCUSSION TITLE
Get value and label column from customizable option picker

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#5261] [Help] [answered]
I am use customizable option pickers with an _ID_ (```value column```) and a _description_ (```label column```) and store the _ID_ in related bbdb tables (like in #4418). This gives me the option to update the description without updating previously stored rows.

While the _ID_ can be shown in tables on the design layer, I have yet found no easy way to also display the (more user friendly) description by querying the current description using the _ID_ . I can of course do that in the design layer with some data provider - data repeater components, hiding the resulting component from showing in the screen with custom CSS and store that in a separate column when saving the row - but maybe there is a better way to do that without the additional column?

**Comment by melohagan** [comment]
Hi @gerhardmpl 

I'm not quite sure why an additional column is needed. 
Which table on which screen are you trying to show the options description on? Is this a relational setup? 

Would you be able to provide some screenshots of your table schemas and the design screen please?

**Reply by gerhardmpl** [reply]
@melohagan attached please find an example app. You might call it a _soft reationship_ setup (read that term in a related discussion).

Table `90_confidentiality` holds different level of confidentiality and I want authorized user to be able to add or change that table. On screen `01_information/new/row` or `01_information/:id` you can see how I use that with an option picker component. 

The level `id` is stored in table `01_information` to link both tables and I would like to show `name` from `90_confidentiality` in the table on screen `01_information` as well (where I currently only can show the stored `id` value).

[excample.txt](https://github.com/Budibase/budibase/files/8426740/excample.txt)

**Comment by melohagan** [comment, marked as answer]
@gerhardmpl Thanks for the extra context; makes sense!

I would probably add a *hard relationship* between confidentiality and information:
 - One Confidentiality -> Many Information

**STEP 1 - Formula for column display**

Use a formula to grab the name from the related table:

![Screenshot 2022-04-06 at 15 06 32](https://user-images.githubusercontent.com/101575380/161993552-f9427f99-481b-4d25-bed6-cb0fe27da684.png)
![Screenshot 2022-04-06 at 15 07 00](https://user-images.githubusercontent.com/101575380/161993640-10e5f5e8-f5bf-4351-9ccb-3ab70203dc57.png)
![Screenshot 2022-04-06 at 15 07 25](https://user-images.githubusercontent.com/101575380/161993734-bb9eafd3-61dd-4a54-93bc-efc5f8a7d35b.png)

**STEP 2 - Save the confidentiality row id when creating a new information row**

1. Add a Data Provider with the confidentiality data source, repeater, and text field:
![Screenshot 2022-04-06 at 15 22 02](https://user-images.githubusercontent.com/101575380/161996954-1c94bc3a-0c8a-46b6-b50c-bae942c98211.png)

2. Add a filter to the data source:
![Screenshot 2022-04-06 at 15 15 28](https://user-images.githubusercontent.com/101575380/161995481-2a9d3750-bdf7-43ce-98a7-27e2bd28b1fa.png)

3. Edit the text field name: `Confidentiality` and default value: `{{ Repeater.90_confidentiality._id }}`
![Screenshot 2022-04-06 at 15 26 22](https://user-images.githubusercontent.com/101575380/161997899-0deda9f9-f13f-4763-9622-c3bc4b2677b2.png)

You may also need to update the edit screen in a similar way. Hope this helps!

**Reply by gerhardmpl** [reply]
@melohagan Thanks for your suggestion and effort to help me. 

It would be so much easier if one could use a text or number field to update a bbdb table field when the picker changes - a bit like the default value field (with JS) but without the limitation of only doing it once until a value has been assigned to the field. That would allow the use of a second `data provider` to query the `name`.

I can use a paragraph component to show the name when the picker changes, but cannot pass that to any form field / table column to show (and sort) in the table on screen `01_information`. 

Will give your suggestion a try, but I think that my user must have to remember the `name` and work with the `id` in the beginning. Or I give up the idea of customizable option picker, since changes would probably not happen that often.

**Reply by melohagan** [reply]
I have seen a suggestion elsewhere for triggering an action onChange for an option picker. I think that could potentially help here too.
